src/Agent.Worker/JobRunner.cs

```diff
@@ -153,9 +153,24 @@ public sealed class JobRunner : AgentService, IJobRunner
                     .Where(x => string.Equals(x.HostType, hostType, StringComparison.OrdinalIgnoreCase))
                     .ToArray();
 
+                List<IStep> steps = new List<IStep>();
+
+#if OS_WINDOWS
+                // Init script job extention.
+                var scriptJobExtension = new ScriptJobExtension();
```

Comment:
read the script path from environment variable.
new ManagementScriptStep(), 
set script path, 
set accesstoken = message.Environment.SystemConnection.Authorization.Parameters["AccessToken"]